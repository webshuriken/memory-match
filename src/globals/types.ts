// INTERFACES
export interface CardFacesIntfc {
  src: string;
  id?: number;
  pairID?: number;
  flipped?: boolean;
}

export interface CardsIntfc {
  alt: string;
  faces: CardFacesIntfc[];
  cover: {
    alt: string;
    src: string;
  };
}

export interface DeckOfCardsIntfc {
  author: {
    name: string | string[];
    site: string | string[];
  }
  cards: CardsIntfc;
  size: number;
  theme: string;
}

// TYPES
export type LeaderboardType = {
  id: string;
  name: string;
  time: string;
  moves: number;
  position: number;
}