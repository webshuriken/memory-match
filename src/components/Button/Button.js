import React from "react";


function Button({bText}) {
  return (
    <button>{bText}</button>
  )
}

export default Button;