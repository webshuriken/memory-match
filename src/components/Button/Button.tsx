import React from "react";


export default function Button({value = 'Button'}): JSX.Element {
  return (
    <button>{value}</button>
  )
}
