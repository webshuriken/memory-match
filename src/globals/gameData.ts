import { LeaderboardType, iDeckOfCardsType } from "./types"
import cardPlaceholderURL from './cards-placeholder.png';


// TODO: FEAT-Game-1: how we expose this data can be changed
// exposes the images public id for the faces and cover.
export const CloudinaryCardsList: { faces: string[]; cover: string } = {
  faces: [
    'cards-blue_hair_uubbsf.png',
    'cards-haku_oxsscv.png',
    'cards-plant_me_thzzz5.png',
    'cards-dragon_plfggm.png',
    'cards-swirl_lady_gagxsr.png',
    'cards-spiral_eye_ueslyj.png',
    'cards-red_hair_f2hosm.png',
    'cards-yuki_greig0.png'
  ],
  cover: 'cards-cover_tusv76.png'
}

// exposes a deck of cards
export const DeckOfCards: iDeckOfCardsType = {
  author: {
    name: [
      'Amy L Sutton',
      'Carlos E Alford'
    ],
    site: [
      'https://amythehatter.com/',
      'https://carlosealford.com/'
    ]
  },
  cards: {
    alt: 'this is a memory card so, no peaking',
    faces: [{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    },{
      src: cardPlaceholderURL
    }],
    cover: {
      alt: 'poker mixed with some old doodles',
      src: cardPlaceholderURL
    },
  },
  size: 8,
  theme: 'poker ala doodles'
}

// exposes game data until we setup a proper backend
export const InitLeaderboard: LeaderboardType[] = [{
  id: 'r33r',
  name: 'Pretender',
  time: '02:44',
  moves: 28,
  position: 1,
},{
  id: 'sc24x',
  name: 'Cerebellum',
  time: '01:28',
  moves: 26,
  position: 3,
},{
  id: 'dfs2',
  name: 'Lindsey',
  time: '00:55',
  moves: 22,
  position: 5,
},{
  id: 'gf2s',
  name: 'Pineapple',
  time: '01:11',
  moves: 0,
  position: 2,
},{
  id: 'jhg56',
  name: '',
  time: '03:33',
  moves: 0,
  position: 4,
},];