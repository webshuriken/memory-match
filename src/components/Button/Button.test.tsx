import renderer from 'react-test-renderer';
import { describe, expect, test } from '@jest/globals';
import Button from './Button';

it('renders a button correctly', () => {
  const component = renderer.create(<Button value="Play" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

// test('renders a button with the value Test', () => {
//   render(<Button value="Test" />);
//   const button = screen.getByRole('button');
//   expect(button).toHaveTextContent('Test');
// });