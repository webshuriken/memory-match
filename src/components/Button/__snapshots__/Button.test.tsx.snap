// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button correctly 1`] = `
<button>
  Play
</button>
`;
